server/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/SourceScoreOrderFragmentsBuilder.java

```diff
@@ -36,11 +35,10 @@
 public class SourceScoreOrderFragmentsBuilder extends ScoreOrderFragmentsBuilder {
 
     private final MappedFieldType fieldType;
-
-    private final QueryShardContext context;
+    private final FetchSubPhase.HitContext context;
 
     public SourceScoreOrderFragmentsBuilder(MappedFieldType fieldType,
-                                            QueryShardContext context,
+                                            FetchSubPhase.HitContext context,
```

Comment:
Maybe just pass the `SourceLookup` directly ?

Reply:
👍